### ISSUE TITLE
Trying to release a booked room crashes the application

### ISSUE THREAD

**Issue by pmhausen** [synyx/MeeRoo#120]
**Tablet:**
Samsung Galaxy Tab A7 Lite
Android 11

**Integration:**
Davx5 connected to CalDAV account
Using dedicated account and personal calendar, not a resource

**Backend:**
SOGo 5.1.0.20210406-1

**How to reproduce:**
With a free room tap "15 Minutes"
Then tap "Release room"
Application crashes

**More info:**
Logged CalDAV transactions during the process
```
Jul 29 13:16:01 sogod [28288]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "PROPFIND /SOGo/dav/fb01/Calendar/personal/ HTTP/1.1" 207 434/290 0.006 1186 63% 0 - 17
Jul 29 13:17:02 sogod [40816]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "PROPFIND /SOGo/dav/fb01/Calendar/personal/ HTTP/1.1" 207 434/290 0.005 1186 63% 0 - 14
Jul 29 13:17:03 sogod [40816]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "PUT /SOGo/dav/fb01/Calendar/personal/b373c694-b4fb-442c-a90a-726f5119d753.ics HTTP/1.1" 201 0/733 0.025 - - 0 - 14
Jul 29 13:17:04 sogod [40816]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "PROPFIND /SOGo/dav/fb01/Calendar/personal/ HTTP/1.1" 207 350/241 0.005 - - 0 - 14
Jul 29 13:17:04 sogod [40816]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "REPORT /SOGo/dav/fb01/Calendar/personal/ HTTP/1.1" 207 402/328 0.006 1763 77% 0 - 14
Jul 29 13:17:05 sogod [40816]: 217.29.33.131, 2a00:b580:8000:11:1c6b:7032:35e9:5616 "PROPFIND /SOGo/dav/fb01/Calendar/personal/ HTTP/1.1" 207 434/290 0.006 1186 63% 0 - 14
```

Thanks and kind regards,
Patrick

**Comment by maxdobler** [comment]
Hi Patrick,

thank you for your Feedback!
We have noticed the same issue on our tablets.

I think it is a bug in the `realse room` logic for CalDav events. I will come back to you if I fixed it.

Again thx for sharing your issue with us, I will try to fix it asap. ;)
Kind regards,
Max

**Comment by heetel** [comment]
Hi Patrick,

we already started to fix this in PR #117 
The app is not crashing anymore, but the calendar is not always being updated (After releasing the room, it's displayed as released on other devices).

I just created a [snapshot release](https://github.com/synyx/MeeRoo/releases/tag/v1.3.2-SNAPSHOT). You can try that out and check if it works for you.

Kind regards,
Julian

**Comment by pmhausen** [comment]
Hi Julian,

seems to work, we noticed that releasing does not remove the appointment from the calendar but changes the ending time to the release time. If that is intentional, we think it's a good idea. So you keep sort of a protocol of bookings.

Kind regards,
Patrick

**Comment by pmhausen** [comment]
Completely removing the app and reinstalling from your APK seems to have improved things. We now get consistent bookings. I'll keep you posted, should that change.

**Comment by heetel** [comment]
We resolved the issue in #117.
Releasing a booked room is now possible for both single events and recurring events.

We observed that sometimes the calendar is not syncing in background, we didn't see today's events in the app until we manually pressed 'sync now' in the system's calendar app.
If you run in the same issue try disabling battery optimization for the calendar app (in the system settings).
